core/src/main/java/tc/oc/pgm/settings/SettingsMatchModule.java

```diff
@@ -0,0 +1,155 @@
+package tc.oc.pgm.settings;
+
+import net.kyori.text.Component;
+import net.kyori.text.TranslatableComponent;
+import net.kyori.text.format.TextColor;
+import net.kyori.text.format.TextDecoration;
+import org.bukkit.Material;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.ClickType;
+import org.bukkit.inventory.ItemFlag;
+import org.bukkit.inventory.ItemStack;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.event.ObserverInteractEvent;
+import tc.oc.pgm.api.setting.SettingKey;
+import tc.oc.pgm.api.setting.SettingValue;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.settings.tools.FlySpeedTool;
+import tc.oc.pgm.settings.tools.GamemodeTool;
+import tc.oc.pgm.settings.tools.NightVisionTool;
+import tc.oc.pgm.spawns.events.ObserverKitApplyEvent;
+import tc.oc.pgm.util.inventory.ItemBuilder;
+import tc.oc.pgm.util.menu.InventoryMenu;
+import tc.oc.pgm.util.menu.InventoryMenuBuilder;
+import tc.oc.pgm.util.menu.InventoryMenuListener;
+import tc.oc.pgm.util.menu.item.InventoryItem;
+import tc.oc.pgm.util.menu.item.InventoryItemBuilder;
+
+@ListenerScope(MatchScope.LOADED)
+public class SettingsMatchModule implements MatchModule, Listener {
```

Comment:
`Settings` are independent of a `Match` so it doesn't make much sense to use a `MatchModule` here. I would also make it a `Listener`, that way we do not need to expose `PGM#getInventoryMenuListener`.

Reply:
I do need to pass around an instance of the listener either way since the implementation needs to query shared data structures that know which player is viewing which inventory menu etc. How should I keep track of that singleton instance outside of PGM?